### DISCUSSION TITLE
Boundary layer parameters extraction in ADFlow

### DISCUSSION THREAD

**Discussion by luigimanni** [mdolab/adflow#335] [Q&A]
Dear all, 

I have performed a CFD analysis on a 2D airfoil on ADflow, and I now need to perform an aeroacoustic evaluation of the airfoil using a Trailing Edge Noise (TEN) code. 

The inputs of the TEN code are the following boundary layer parameters, extracted text to the trailing edge:

- BL thickness (99%)
- BL displacement thickness
- BL momentum thickness
- Momentum thickness Reynolds number

Can anyone suggest me how to extract these properties from the simulation on ADFLOW? 

Thank you very much in advance.

**Comment by hntter** [comment]
Hi, I met the same problem as yours. Have you find ways how to extract boundary layer parameters?

**Comment by GodotMisogi** [comment]
As shown in the figures below, you can trace a line along the `pyHyp` grid. Note that this is a mesh-dependent procedure that inherently depends on the thin-shear layer definitions for the boundary layer related to the surface's curvature.
![fields_doc](https://github.com/user-attachments/assets/d7dc6358-4324-4b34-ac18-3a713f552da4)
![bl_plot](https://github.com/user-attachments/assets/7ea9e477-6efa-4804-ae4e-ebef59baf9cb)
